### PR TITLE
[WIP] Include simCLR

### PR DESCRIPTION
**Pull request by alexrey88**
The idea is to implement simCLR in the code: train the base encoder with simCLR, and then fine-tune the weights of the encoder with our main task (the masker). See if we get better results than with current deeplabv2 encoder.

### CODE REVIEW

File: omnigan/data.py [hunk 1]
```diff
@@ -209,3 +236,31 @@ def get_all_loaders(opts):
...
+
+def get_simclr_loaders(opts):
+    loaders = {}
+    for mode in ["train", "val"]:
+        loaders[mode] = {}
+        for domain in ["r", "s"]:
+            if mode in opts.data.files:
+                if domain in opts.data.files[mode]:
+                    if opts.gen.simclr.domain_adaptation or (
+                        not opts.gen.simclr.domain_adaptation and domain == "r"
```
Comment (alexrey88) [inline omnigan/data.py:263]: remove not part

**Comment by alexrey88** [comment]
Reminder : add art option for pretraining, and moves simclr loss computation to another function get_pretraining_loss
